File: Harish Kumar/Dyanamic Programming & Recursion /Min Cost Climbing Stairs/min costclimbing stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        
        int min1 = cost[0];
        int min2 = cost[1];
        
        int min = 0;
        
        for(int i = 2;i < cost.length;i++)
        {
            min = cost[i] + Math.min(min1 ,min2);
            
            min1 = min2;
            min2 = min;
        }
        return Math.min(min1,min2);
    }
}

File: Aman/DP Recursion and Backtracking/Min Cost Climbing Stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int m1=cost[0],m2=cost[1],m,n=cost.size();
        for(int i=2;i<n;i++)
        {
            m=cost[i]+min(m1,m2);
            m1=m2;
            m2=m;
        }
        return min(m1,m2);
    }
};
